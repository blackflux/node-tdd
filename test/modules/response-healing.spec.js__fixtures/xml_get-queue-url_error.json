
{
  "fn": "call",
  "params": [
    "sqs:GetQueueUrlCommand",
    {"QueueName": "some-queue-name"},
    {"expectedErrorCodes": ["QueueDoesNotExist"]}
  ],
  "cassette": [
    {
      "scope": "https://sqs.us-west-2.amazonaws.com:443",
      "method": "POST",
      "path": "/",
      "body": "QueueName=some-queue-name&Action=GetQueueUrl&Version=2012-11-05",
      "status": 400,
      "response": "<?xml version=\"1.0\"?><ErrorResponse xmlns=\"http://queue.amazonaws.com/doc/2012-11-05/\"><Error><Type>Sender</Type><Code>AWS.SimpleQueueService.NonExistentQueue</Code><Message>The specified queue does not exist for this wsdl version.</Message><Detail/></Error><RequestId>7fb23608-cc30-515b-89b2-6c7fbfb18a2c</RequestId></ErrorResponse>",
      "responseIsBinary": false
    }
  ]
}
