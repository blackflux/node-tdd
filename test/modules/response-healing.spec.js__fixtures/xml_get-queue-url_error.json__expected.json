[
  {
    "scope": "https://sqs.us-west-2.amazonaws.com:443",
    "method": "POST",
    "path": "/",
    "body": {
      "QueueName": "some-queue-name"
    },
    "status": 400,
    "response": {
      "__type": "com.amazonaws.sqs#QueueDoesNotExist",
      "message": "The specified queue does not exist."
    },
    "responseIsBinary": false,
    "reqheaders": {
      "amz-sdk-invocation-id": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$",
      "amz-sdk-request": "attempt=1; max=3",
      "authorization": "^AWS4-HMAC-SHA256 Credential=.+$",
      "content-length": "^[1-9][0-9]*$",
      "content-type": "application/x-amz-json-1.0",
      "host": "sqs.us-west-2.amazonaws.com",
      "user-agent": "^aws-sdk-js/.+$",
      "x-amz-content-sha256": "^[a-f0-9]{64}$",
      "x-amz-date": "^[0-9]{8}T[0-9]{6}Z$",
      "x-amz-target": "AmazonSQS.GetQueueUrl",
      "x-amz-user-agent": "^aws-sdk-js/[0-9]+.[0-9]+.[0-9]+$",
      "x-amzn-query-mode": "true",
      "connection": "close"
    }
  }
]
